nftables.nix

```diff
@@ -23,87 +23,51 @@ let cfg = config.pi-router.interfaces; in
-              ct state {established, related} accept
-
-              # ICMP
-              # routers may also want: mld-listener-query, nd-router-solicit
-              ip6 nexthdr icmpv6 icmpv6 type { destination-unreachable, packet-too-big, time-exceeded, parameter-problem, nd-router-advert, nd-neighbor-solicit, nd-neighbor-advert } accept
-              ip protocol icmp icmp type { destination-unreachable, router-advertisement, time-exceeded, parameter-problem } accept
-
-              # allow "ping"
-              ip6 nexthdr icmpv6 icmpv6 type echo-request accept
-              ip protocol icmp icmp type echo-request accept
-
-              # allow SSH
-              tcp dport 22 accept
-
-              # allow DNS via LAN
-              iifname $LAN_IFC tcp dport {53} accept
-              iifname $LAN_IFC udp dport {53} accept
-
-              # count and drop any other traffic
-              counter drop
-          }
-
-          # Allow all outgoing connections.
-          chain output {
-              type filter hook output priority 0;
-              accept
-          }
-
-          chain forward {
-              type filter hook forward priority 0;
-              
-              # allow LAN to WAN
-              iifname $LAN_IFC oifname $WAN_IFC accept
-
-              # drop new packages between interfaces
-              iifname $ALL_IFC oifname $ALL_IFC ct state new counter drop
-
-              accept
-          }
-      }
-
-      table ip nat {
-          chain PREROUTING {
-              type nat hook prerouting priority dstnat; policy accept;
-          }
-
-          chain INPUT {
-              type nat hook input priority 100; policy accept;
-          }
-
-          chain OUTPUT {
-              type nat hook output priority -100; policy accept;
-          }
+  networking = {
+
+    firewall.enable = true;
+    nftables.enable = true;
+
+    nat = {
+      enable = true;
+      externalInterface = cfg.wan.name;
+      internalInterfaces = [ cfg.lan.name ];
+    };
+
+    firewall.extraInputRules = ''
+      # ICMP:
+      # routers may also want: mld-listener-query, nd-router-solicit
+      ip6 nexthdr icmpv6 icmpv6 type {
+        destination-unreachable,
+        packet-too-big,
+        time-exceeded,
+        parameter-problem,
+        nd-router-advert,
+        nd-neighbor-solicit,
+        nd-neighbor-advert
+      } accept
+
+      ip protocol icmp icmp type {
+        destination-unreachable,
+        router-advertisement,
+        time-exceeded,
+        parameter-problem
+      } accept
+
+      # count and drop any other traffic
+      counter drop
+    '';
 
-          chain POSTROUTING {
-              type nat hook postrouting priority srcnat; policy accept;
+    firewall.filterForward = true;
 
-              # NAT between LAN and WAN
-              oifname $WAN_IFC ip saddr $LAN_NET counter masquerade
-          }
-      }
+    firewall.extraForwardRules = ''
+      # type filter hook forward priority 0;
```

Comment:
Can we include that without it failing?

Reply:
For some reason we can't. Wasn't a problem with my old config... 🤔

Reply:
No idea why. Let's leave it commented out for now.